crates/ruma-common/src/identifiers/room_or_alias_id.rs

```diff
@@ -12,6 +12,8 @@ use super::{server_name::ServerName, OwnedRoomAliasId, OwnedRoomId, RoomAliasId,
 /// `RoomOrAliasId` is useful for APIs that accept either kind of room identifier. It is converted
 /// from a string slice, and can be converted back into a string as needed. When converted from a
 /// string slice, the variant is determined by the leading sigil character.
+/// `RoomOrAliasId` can also be converted to either `RoomId` or `RoomAliasId` by matching against
```

Comment:
This is treated as markdown, for a new paragraph you need to leave an empty line:
```suggestion
///
/// `RoomOrAliasId` can also be converted to either `RoomId` or `RoomAliasId` by matching against
```

I feel like the message we are trying to convey is still not clear. Maybe adding a second sentence like: "When using `match` on the result, `Ok(_)` will give one type and `Err(_)` will give the other."